src/dune/__main__.py

```diff
@@ -18,16 +19,63 @@ def handle_simple_args():
         handle_version()
         sys.exit(0)
 
+def load_module(absolute_path):
+    module_name, _ = os.path.splitext(os.path.split(absolute_path)[-1])
+    module_root = os.path.dirname(absolute_path)
+
+    sys.path.append(module_root)
+    spec = importlib.util.spec_from_file_location(module_name, absolute_path)
+    py_mod = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(py_mod)
+    return py_mod
+
+def load_all_modules_from_dir(plugin_dir):
+    loaded_modules = []
+
+    if not os.path.exists(plugin_dir):
+        return loaded_modules
+
+    for subdir in os.listdir(plugin_dir):
+        subdir_path = os.path.join(plugin_dir, subdir)
```

Comment:
We probably should NOT check for `main.py` when `os.path.isdir(subdir_path)` returns false. Instead why don't we warn the user *if* the file has a `.py` extension?

This way we can add a `README.md` file to `src/plugin` - which is something we should also do.

Reply:
fixed checking `isdir` and added README.md.
additionally added docs/ and README.md to the package